lib/linguist/heuristics.yml

```diff
@@ -426,6 +426,19 @@ disambiguations:
     pattern: '^\s*(import.+(from\s+|require\()[''"]react|\/\/\/\s*<reference\s)'
   - language: XML
     pattern: '(?i:^\s*<\?xml\s+version)'
+- extensions: ['.v']
+  rules:
+  - language: Coq
+    pattern:
+    - '^(Proof\.|Qed\.$|Theorem\s+[\w'']+:|Inductive\s+\w+)'
+    - '^Require\s+(Import|Export)(\s+("\w+"|\w+))+\.'
+    - '(?m:^Module\s+\w+\..*?End\s+\w+\.)'
+  - language: Verilog
+    pattern:
+    - '^\s*module\s+\w+(\(|;$)'
+    - '^\s*endmodule\s*(//.*)?$'
+    - '^\s*endfunction\s*(//.*)?$'
```

Comment:
I'm worried about the performance impact of some of these regular expressions (the last Coq pattern and these two Verilog patterns). They will result in a lot of backtracking.

Maybe we should try without the last Coq pattern? Could we also reduce these two Verilog patterns to match on `^\s*(endfunction|endmodule)\s`? Wouldn't that be sufficient?